daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-route-messages.md

```diff
@@ -0,0 +1,246 @@
+---
+type: docs
+title: "How-To: Route messages to different event handlers"
+linkTitle: "How-To: Route events to different handlers"
+weight: 2100
+description: "Learn how to route messages from a topic to different event handlers based on CloudEvent fields"
+---
+
+## Introduction
+
+[Content-based routing](https://www.enterpriseintegrationpatterns.com/ContentBasedRouter.html) is a messaging pattern that utilizes a DSL instead of imperative application code. PubSub Routing is an implementation of this pattern that allows developers to use expressions to route [CloudEvents](https://cloudevents.io) based on their contents to different URIs/paths and event handlers in your application. If no route matches, then an optional default route is used. This becomes useful as your applications expands to support multiple event versions, or special cases. Routing can be implemented with code; however, keeping routing rules external from the application can improve portability.
```

Comment:
```suggestion
[Content-based routing](https://www.enterpriseintegrationpatterns.com/ContentBasedRouter.html) is a messaging pattern that utilizes a DSL instead of imperative application code. PubSub routing is an implementation of this pattern that allows developers to use expressions to route [CloudEvents](https://cloudevents.io) based on their contents to different URIs/paths and event handlers in your application. If no route matches, then an optional default route is used. This becomes useful as your applications expands to support multiple event versions, or special cases. Routing can be implemented with code; however, keeping routing rules external from the application can improve portability.
```